internal/db/schema/migrations/oss/postgres/99/02_worker_authentication.up.sql

```diff
@@ -0,0 +1,147 @@
+begin;
+
+create table worker_auth_operational_indicator_enm(
+  state text primary key
+    constraint only_predefined_worker_auth_states_allowed
+      check (
+          state in ('next','current')
+        )
+);
+
+comment on table worker_auth_operational_indicator_enm is
+  'worker_auth_operational_indicator_enm is an enumeration table representing the state of certificates used for worker authentication.';
+
+insert into worker_auth_operational_indicator_enm (state)
+values
+  ('next'),
+  ('current');
+
+create table worker_auth_ca_certificate(
+ serial_number numeric primary key,
```

Comment:
This can't be a primary key. The reason is that serial numbers must be unique per CA certificate, but that fact has some nuances:

1. Not all CAs are conforming and mistakenly issue certs with the same serial number, especially for the same public key (which they shouldn't)
2. A CA cert that is self signed can be _any serial number it wants_ including the same serial number as another self signed CA -- quite often just `1`.
3. Even across CAs serial numbers can repeat; it's not uncommon for a given issuer to not use random serials but rather start at `1` and go up sequentially.

The correct way to ensure uniqueness -- ignoring option #1 above -- would be to have the primary key be (public key, serial number) tuple. In this particular case I would think you could simply use `state` or barring that `public_key`.